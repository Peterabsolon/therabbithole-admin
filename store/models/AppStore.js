import { types } from 'mobx-state-tree'
import { SourcesStore } from './SourcesStore'
import { RouterStore } from './RouterStore'
import { ArticlesStore } from './ArticlesStore'

// import storage from 'store/helpers/storage'

export const AppStore = types.model(
  'AppStore',
  {
    title: types.string,
    isLoading: types.optional(types.boolean, true),
    sourcesStore: types.optional(SourcesStore, {
      sources: [],
    }),
    articlesStore: types.optional(ArticlesStore, {
      list: [],
    }),
    router: types.optional(RouterStore, {
      pathname: '',
      params: '',
    }),
  },
  {
    markLoading(loading) {
      this.isLoading = loading
    },
    afterCreate() {
      // if (process.env.NODE_ENV === 'production') {
      //   storage.persist(this)
      // }
    },
  }
)
