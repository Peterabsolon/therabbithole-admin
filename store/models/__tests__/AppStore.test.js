// import { when } from 'mobx'
import { AppStore } from '../AppStore'

// const bookFetcher = () => Promise.resolve(JSON.parse(fs.readFileSync("./public/books.json")))

// it('bookstore fetches data', () => {
//   const store = AppStore.create({ title: 'App title' })
//   console.log('created store', store.sourcesStore.sources.length)
//   expect(store.sourcesStore.sources.length).toBe(0)
//   //   when(
//   //     () => store.isLoading === false,
//   //     () => {
//   //       expect(store.sourcesStore.sources.length).toBe(0)
//   //       done()
//   //     }
//   //   )
// })
describe('ðŸ—„ App Store', () => {
  it('Init', () => {
    const store = AppStore.create({ title: 'Mobx Starter kit' })
    expect(store).toMatchSnapshot()
  })
})
