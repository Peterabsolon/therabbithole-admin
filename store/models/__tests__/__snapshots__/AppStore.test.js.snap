// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ—„ App Store Init 1`] = `
Object {
  "articlesStore": Object {
    "list": Array [],
  },
  "isLoading": true,
  "router": Object {
    "params": "",
    "pathname": "",
  },
  "sourcesStore": Object {
    "isLoading": true,
    "sources": Array [],
  },
  "title": "Mobx Starter kit",
}
`;
